backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs

```diff
@@ -115,4 +119,59 @@ public async Task<LessonDto> CreateLessonAsync(NewLessonDto lessonDto)
 
         return _mapper.Map<LessonDto>(createdLesson);
     }
+
+    public async Task<TeacherStatisticsDto> GetTeacherLessonsStatisticsAsync()
+    {
+        long id = _authService.UserId;
+
+        var statistics = new TeacherStatisticsDto();
+
+        statistics.TotalClasses = await _context.Lessons.CountAsync(l => l.CreatedBy == id);
+        statistics.CanceledClasses = await _context.Lessons.CountAsync(l => l.CreatedBy == id && l.IsCanceled);
+        statistics.FutureClasses = await _context.Lessons.CountAsync(l => l.CreatedBy == id && l.StartAt > DateTime.UtcNow && !l.IsCanceled);
+        var total = await _context.Lessons.Where(l => l.CreatedBy == id && l.StartAt < DateTime.UtcNow).Select(l => l.Subscribers.Count).ToListAsync();
+        if (total is not null)
+        {
+            statistics.TotalStudents = total.Sum();
+        }
+        statistics.NextClass = await _context.Lessons.Where(l => l.CreatedBy == id && l.StartAt > DateTime.UtcNow && !l.IsCanceled)
+                                                     .OrderBy(l => l.StartAt)
+                                                     .Select(l => l.StartAt)
+                                                     .FirstOrDefaultAsync();
+        if (statistics.NextClass == DateTime.MinValue) 
+        {
+            statistics.NextClass = null;
+        }
+        return statistics;
+    }
+
+    public async Task<ICollection<DaysWithLessonsDto>> GetLessonsInPeriodAsync(DateTime start, DateTime end)
+    {
+        long id = _authService.UserId;
+        var selectedLessons = await _context.Lessons.Where(l => l.CreatedBy == id && l.StartAt > start && l.StartAt < end)
+                                                    .Include(l => l.Tags)
+                                                    .OrderBy(l => l.StartAt)
+                                                    .ToListAsync();
+                                                    
+        var groupedLessons = selectedLessons.GroupBy(l => l.StartAt.Date)
+                                            .Select(d => new DaysWithLessonsDto { LessonsDate = d.Key, Lessons = _mapper.Map<List<Lesson>, List<LessonDto>>(d.ToList()) })
+                                            .ToList();
+
+        return groupedLessons;
+    }
```

Comment:
I guess it also can be one IQueryable request

Reply:
This one I told you about. If we cast groupBy and some data modifications in one request it causes an error. So the only way to avoid the error is to use a list in between iqueryable and groupby